receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -182,10 +217,9 @@ func (kr *k8sobjectsreceiver) startWatch(ctx context.Context, config *K8sObjects
 			}
 		case <-stopperChan:
 			watcher.Stop()
-			return
+			return true
 		}
 	}
-
 }
 
 func getResourceVersion(ctx context.Context, config *K8sObjectsConfig, resource dynamic.ResourceInterface) (string, error) {
```

Comment:
It would necessitate an extra API call, but could we call `resource.List` with `config.ResourceVersion` here (if provided) to validate that the version is valid before returning it? We could then just return the latest version if it isn't valid. I think that would simplify the code since we could contain all the necessary changes to this function.

Reply:
@evan-bradley we could do that for the `resourceVersion` passed in via the config, but during operation it is possible we'll get a 410, resulting in no resource version, and have to do a clean list without a resource version to get a new one.  I think it ends up being simpler if we combine the 2 use cases instead of separating out the checking of the initial resource vs 410 responses.

Reply:
You're right, it looks like it is on us to handle the 410 during operation. I was under the impression that the `RetryWatcher` handled this case, but it doesn't. Is there a reason we couldn't use a [Reflector](https://pkg.go.dev/k8s.io/client-go@v0.28.1/tools/cache#Reflector) to watch objects while handling resource version expiry? My gut reaction is that the k8s Go client should handle this behavior.

Reply:
I went looking for answers in the Reflector api and could find something that does Dynamic calls like we need or handles reconnecting like the retrywatcher.

Reply:
Makes sense, thanks.